src/pages/Lobby.tsx

```diff
@@ -1,10 +1,48 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import WebSocketConnection from '../modules/ws/websocket';
+import { UsersJSON, GamesJSON } from '../types';
+
+type LobbyProps = {
+  ws: WebSocketConnection | null;
+};
+
+const PLACEHOLDER_USERS = [{ username: 'user' }, { username: 'user1' }, { username: 'user2' }];
+const PLACEHOLDER_GAMES = [{ game: 'game' }, { game: 'game1' }, { game: 'game2' }];
+
+const LobbyPage = ({ ws }: LobbyProps): JSX.Element => {
+  /**
+   * States
+   */
+  const [users, setUsers] = useState<Record<string, unknown>[]>([]);
+  const [games, setGames] = useState<Record<string, unknown>[]>([]);
+
+  const updateUsers = (payload: unknown) => {
+    const data = payload as UsersJSON;
+    const { users: newUsers } = data;
+    setUsers(newUsers);
+  };
+
+  const updateGames = (payload: unknown) => {
+    const data = payload as GamesJSON;
+    const { games: newGames } = data;
+    setGames(newGames);
+  };
+
+  useEffect(() => {
+    ws?.sendMessage('GAMES', {});
+    ws?.sendMessage('USERS', {});
+
+    ws?.addListener('GAMES', updateUsers);
+    ws?.addListener('USERS', updateGames);
```

Comment:
use if check to avoid optional chaining:
```
if (ws) {
    ws.sendMessage('GAMES', {});
    ws.sendMessage('USERS', {});

    ws.addListener('GAMES', updateUsers);
    ws.addListener('USERS', updateGames);
}
```

Reply:
Done